### PR TITLE
Add and remove player commands

### PR DESCRIPTION
**Pull request by Derekholio**

### CODE REVIEW

File: src/SrBot.ts [hunk 1]
```diff
@@ -197,12 +199,54 @@ export class SrBot {
...
                 break;
         }
     }
 
+    private addPlayer(message : Discord.Message, username: string) : boolean {
+        const serverId = message.member.guild.id;
+        return this.statsGenerator.addPlayer(serverId, username);
+    }
+
+    private removePlayer(message : Discord.Message, username : string) : boolean {
```
Comment (rypete) [inline src/SrBot.ts]: Are these `addPlayer`/`removePlayer` methods needed in this class?  Could we call just call `this.statsGenerator.###Player(serverId, username);` where they are used?

File: src/SrBot.ts [hunk 2]
```diff
@@ -197,12 +199,54 @@ export class SrBot {
...
+        const serverId = message.member.guild.id;
+        return this.statsGenerator.removePlayer(serverId, username);
+    }
+
+    private isValidUsername(username: string): boolean {
```
Comment (rypete) [inline src/SrBot.ts]: Maybe make this into a util function.  It doesnt use anything explicitly in this class

File: src/StatsGenerator.ts [hunk 3]
```diff
@@ -90,6 +100,41 @@ export class StatsGenerator {
         }
     }
 
+    public addPlayer(serverId: string, username: string) : boolean {
+        const server = this.getServer(serverId);
+        if (server){
+            const player = {SR: 0, player: username, private: false} as Player;
+            server.players.push(player);
```
Comment (rypete) [inline src/StatsGenerator.ts:107]: I think I started this fire but we gotta stop mutating the lastresult object (which `getServer` returns).  I'm ok with it in this PR but flagging we need to tech debt a proper implementation to update server properties

File: src/SrBot.ts (hunk 1 above)
Comment (Derekholio) [reply src/SrBot.ts]: Done

File: src/SrBot.ts (hunk 2 above)
Comment (Derekholio) [reply src/SrBot.ts]: Done

File: src/StatsGenerator.ts (hunk 3 above)
Comment (Derekholio) [reply src/StatsGenerator.ts:107]: Trusty ole tech debt

**Review by rypete — APPROVED** [review]

**Pull request merged**